ghcide/src/Development/IDE/Main.hs

```diff
@@ -206,14 +206,14 @@ defaultMain Arguments{..} = do
                 t <- t
                 hPutStrLn stderr $ "Started LSP server in " ++ showDuration t
 
-                dir <- IO.getCurrentDirectory
+                dir <- maybe IO.getCurrentDirectory return rootPath
 
                 -- We want to set the global DynFlags right now, so that we can use
                 -- `unsafeGlobalDynFlags` even before the project is configured
                 -- We do it here since haskell-lsp changes our working directory to the correct place ('rootPath')
                 -- before calling this function
```

Comment:
I think this comment needs to be updated, though it still will be true until we update `lsp`.

Also, `fromMaybe` in L220 can be simplified.

Reply:
good catch, thanks!